File: LeetCode/cpp/74.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>> &matrix, int target) {
        int m = matrix.size(), n = m == 0 ? 0 : matrix[0].size();
        if (m == 0)
            return false;
        int i = m - 1, j = 0;
        while (j < n && i >= 0) {
            if (matrix[i][j] > target)
                --i;
            else if (matrix[i][j] < target)
                ++j;
            else
                return true;
        }
        return false;
    }
};

File: LeetCode/python3/74.py
class Solution:
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        m,n = len(matrix), len(matrix[0]) if len(matrix) > 0 else 0
        if m == 0:
            return False
        i , j = m - 1, 0
        while i >= 0 and j < n:
            if target > matrix[i][j]:
                j += 1
            elif target < matrix[i][j]:
                i -= 1
            else:
                return True
        return False